src/daemon/rpmostreed-os.c

```diff
@@ -77,6 +81,101 @@ sysroot_changed (RpmostreedSysroot *sysroot,
     g_warning ("%s", local_error->message);
 }
 
+static gboolean
+os_authorize_method (GDBusInterfaceSkeleton *interface,
+                     GDBusMethodInvocation  *invocation)
+{
+  RpmostreedOS *self = RPMOSTREED_OS (interface);
+  const gchar *method_name = g_dbus_method_invocation_get_method_name (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  GVariant *parameters = g_dbus_method_invocation_get_parameters (invocation);
+  const gchar *action = NULL;
+  gboolean authorized = FALSE;
+
+  if (g_strcmp0 (method_name, "GetDeploymentsRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedRebaseRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadRebaseRpmDiff") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.repo-refresh";
+    }
+  else if (g_strcmp0 (method_name, "Deploy") == 0 ||
+           g_strcmp0 (method_name, "Rebase") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.rebase";
```

Comment:
Hmm, I'm thinking now it'd make more sense to put `Deploy` together with `Upgrade`. Both of them are constrained to stay within the same branch (after https://github.com/projectatomic/rpm-ostree/pull/495), whereas `Rebase` is much more powerful.

`Deploy` can also be used as a safer race-free version of `Upgrade`, as used by Cockpit (and probably gnome-software should use `Deploy` as well here if e.g. pending versions/pkg diff previews etc... are displayed).